launch/test/launch/substitutions/test_command.py

```diff
@@ -29,9 +29,9 @@ def commands():
     this_dir = pathlib.Path(__file__).parent
 
     commands = {
-        'normal': str(this_dir / 'test_command' / 'normal_command.bash'),
-        'failing': str(this_dir / 'test_command' / 'failing_command.bash'),
-        'with_stderr': str(this_dir / 'test_command' / 'command_with_stderr.bash')
+        'normal': this_dir / 'test_command' / 'normal_command.bash',
+        'failing': this_dir / 'test_command' / 'failing_command.bash',
+        'with_stderr': this_dir / 'test_command' / 'command_with_stderr.bash'
     }
```

Comment:
I think this still needs to be converted to `str`, because we're calling `value.replace('bash', 'bat')` with them below. `Path.replace()` is not the same thing as `str.replace()`. It will at least not silently fail (but only on Windows, see the `if` below) because the number of args is different: https://docs.python.org/3/library/pathlib.html#pathlib.Path.replace.

```
E   TypeError: Path.replace() takes 2 positional arguments but 3 were given
```

or you could adapt the `replace()` call itself.

Reply:
I've changed `.replace('bash', 'bat')` to `.with_suffix('bat')` which uses the pathlib semantics more nicely. I assume this was going to be caught by the CI run before merge, but it's nice to get it beforehand, good eye as always.